libs/common/src/adaptors/error-handling-adaptor.ts

```diff
@@ -0,0 +1,30 @@
+import { NextFunction, Request, Response } from 'express';
+
```

Comment:
Open to suggestions -- I'd say its an adaptor to make async funcs adaptable to express 4 error handling -- it returns a a wrapped function. It's not really middleware

Reply:
It accepts req/res/next and will be called in routes - that is middleware?

https://github.com/UK-Export-Finance/dtfs2/pull/3920/files#diff-1666f970c445a448a3e341e9812a911839bbf921974096793922fcfbc7687f7c

Reply:
this accepts a function and wraps an existing route, returning a function wrapped in code that makes it so our error handler works (although this was working, and now isnt -- so this conversation might be redundant).

Looking at express, it refers to every function of the route as middleware, so `roleBlaAccessControl(maker)`, `validateBla`, `blaController.getBla` are all middleware according to express -- so technically what `errorHandlingAdaptor` is doing is taking in a middleware as the only parameter, wrapping it and returning this wrapped function (which I guess is middleware?)

Reply:
`wrapMiddlewareWithErrorhandlingWrapper`?

Reply:
If it is a middleware can we place this inside `middleware` function?

Reply:
It's not middleware -- its a wrapper, it's not accepting `res/req/next` as a parameter

Reply:
I really don't see how this is _not_ middleware - it consumes req/res/next and will be consumed in routes, alongside other existing middleware.

It would be strange to have this called anything other than middleware and IMO would otherwise be confusing